### PR TITLE
Better gsutil error reporting

### PR DESCRIPTION
**Pull request by jacoblurye**
...and cleaner stdout feedback communication in general.

### CODE REVIEW

File: cli/api.py [hunk 1]
```diff
@@ -33,7 +33,15 @@ def _error_message(response: requests.Response):
         if response.status_code >= 500:
             message = f"API server error: {message}"
         if type(message) == dict and "errors" in message:
-            return "Multiple errors:\n  " + "\n  ".join(map(str, message["errors"]))
+            message_lines = ["Multiple errors:"]
+            message_lines.extend(
+                [
+                    click.style("* ", dim=True)
+                    + click.style(str(message), fg="bright_red")
```
Comment (curlup) [inline cli/api.py:40]: nit: maybe don't make all the text bright red? what if only `*` were red?

File: cli/upload.py [hunk 2]
```diff
@@ -159,43 +165,42 @@ def _start_procs(upload_info: api.UploadInfo, xlsx: str) -> list:
...
                 if p.returncode != 0:
                     errored = True
 
-            errline = p.stderr.readline()
+            click.echo(p.args[-2])
 
-            # skipping "large file" warnings
-            if errline.strip() in _IGNORED_WARN_LINES:
-                continue
+        time.sleep(1)
```
Comment (curlup) [inline cli/upload.py:195]: why is that needed here?

File: cli/upload.py [hunk 3]
```diff
@@ -159,43 +165,42 @@ def _start_procs(upload_info: api.UploadInfo, xlsx: str) -> list:
...
-                    errline = errline.split("]", 1)[1]
-                print(
-                    f"[{len(finished)}/{len(procs)} done] {i+1}: {p.args[-2]}: "
-                    + errline,
-                    end="",
-                )
-
-    print(f"[{len(finished)}/{len(procs)} done]")
+    if errored:
+        click.secho(f"!!! encountered upload error !!!", fg="bright_yellow", bold=True)
```
Comment (curlup) [inline cli/upload.py:201]: red maybe?

File: cli/upload.py [hunk 4]
```diff
@@ -159,43 +165,42 @@ def _start_procs(upload_info: api.UploadInfo, xlsx: str) -> list:
...
+    progress_bar = "."
     while len(finished) != len(procs) and not errored:
-        for i, p in enumerate(procs):
 
+        click.secho(f"Uploading files ({len(finished)} / {len(procs)} done)", bold=True)
+        click.echo(progress_bar)
+        for i, p in enumerate(procs):
+            click.secho(f"(proc {i + 1}) ", nl=False, fg="bright_blue")
             if i in finished:
+                click.secho("uploaded ", nl=False, fg="green", bold=True)
```
Comment (curlup) [inline cli/upload.py:182]: nit: so many colors!

File: cli/upload.py [hunk 5]
```diff
@@ -159,43 +165,42 @@ def _start_procs(upload_info: api.UploadInfo, xlsx: str) -> list:
...
 
-            if errline and len(errline) > 2:  # skipping '* ' spinner lines
+        progress_bar += "."
 
-                # changing output from always "[0/1 files]" - due to manual
-                # parallelization of gsutil to "[x/y files]"
-                if errline.split("]", 1)[0].endswith("/1 files"):
-                    errline = errline.split("]", 1)[1]
-                print(
-                    f"[{len(finished)}/{len(procs)} done] {i+1}: {p.args[-2]}: "
```
Comment (curlup) [inline cli/upload.py:193]: this reported progress for large files, like
```
[85/101 done] 77: [ 1.2 GiB/ 3.2 GiB]  38% Done
[85/101 done] 77: [ 1.9 GiB/ 3.2 GiB]  59% Done
...
[85/101 done] 77: [ 3.1 GiB/ 3.2 GiB]  99% Done
```
why are we removing those? I'd think showing progress and its speed is good and should not be removed.
Open for discussion

File: tests/test_upload.py [hunk 6]
```diff
@@ -211,6 +212,7 @@ def stdout():
     failed = MagicMock()
     failed.return_value = api.MergeStatus("upload-failed", "some error details", None)
     monkeypatch.setattr(api, "poll_upload_merge_status", failed)
+    print("*******")
```
Comment (curlup) [inline tests/test_upload.py:215]: 🤔

File: cli/upload.py (hunk 5 above)
Comment (jacoblurye) [reply cli/upload.py:193]: Oh, I didn't know that - should've checked with you first! That's definitely something we should keep.

File: cli/upload.py (hunk 2 above)
Comment (jacoblurye) [reply cli/upload.py:195]: Since we print every time we poll, this reduces the amount of output the CLI generates. Got any arguments against `sleep`ing? I don't feel strongly.

File: cli/upload.py (hunk 5 above)
Comment (jacoblurye) [reply cli/upload.py:193]: Wait, actually - won't the addition on line 193 still print out progress for large files? All I removed was reporting `[{len(finished)}/{len(procs)} done]` on every line (now it's reported once for every process polling iteration: see line 177).

File: cli/upload.py (hunk 2 above)
Comment (curlup) [reply cli/upload.py:195]: I don't see how sleeping reduces output? 
1. Pool will sleep for until there's an output to pass
2. We still need to print everything we get

so i think this just hangs printing for a second, not changing other behavior - thus the question why would we need it

File: cli/upload.py [hunk 7]
```diff
@@ -163,45 +163,54 @@ def _start_procs(upload_info: api.UploadInfo, xlsx: str) -> list:
...
                 if p.returncode != 0:
-                    errored = True
-
-            click.echo(p.args[-2])
+                    message += click.style(
+                        f"!!! upload error !!! ", fg="red", bold=True
+                    )
+                    message += p.args[-2]
+                    click.echo(message)
+                    error = f"{prev_errlines[i]}{errline}"
```
Comment (curlup) [inline cli/upload.py:196]: not sure what's happening here

Comment (jacoblurye) [reply cli/upload.py:196]: I'll add a comment that will hopefully clarify.
